packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java

```diff
@@ -106,7 +108,10 @@ private boolean isStringKeyedMap(Object value) {
 
   @Override
   public void onMethodCall(MethodCall call, Result result) {
+    Context context =
```

Comment:
Would this logic be a candidate for a `Context activeContext()` or similarly named method on `Registrar`?

Reply:
Yes. Like so https://github.com/alibitek/engine/commit/4f26a925b934d97befc999e708d5bb96703f966e ?

Reply:
I think it would be useful to have an `activeContext()` method in the Registrar itself, as the way to obtain a context is common among plugins.
It would reduce the need to repeat this code. What do you think?

In the meantime, I've added an activeContext method to the `android_intent` plugin.
Should I proceed with submitting the activeContext to the flutter/engine project? Any other changes there? 
On iOS, the FlutterPluginRegistrar protocol doesn't have the same functionality, so no changes there.

Reply:
@alibitek I think it would make sense to add it, if

1. Plugin authors often find themselves in a situation where they need a context and where it should be the current `Activity`, if one exists, but can live with the `Application`, if not; and
1. We can't just make the existing `context()` method return the active context---presumable because other callers need the returned context to be the `Application`.

If you think both of the above apply, then we'd much appreciate you moving forward with a `flutter/engine` PR.

Adding an `activeContext` method to the `Registrar` interface is a breaking change whose impact is limited since we provide the default implementation of that interface ourselves. Still, we need to announce it to `flutter-dev`.

Alternatively, we could create a utility class for such extensions:
```java
public final class RegistrarExt implements Registrar {
  private final Registrar registrar;
  public RegistrarExt(Registrar registrar) {
    this.registrar = registrar;
  }
  public Context activeContext() {
    final Activity activity = registrar.activity();
    return (activity == null) ? registrar.context() : activity;
  }
  // + delegate implementations of Registrar methods.
}
This would make sense if we expect additional extension methods to accrue over time. A more modern alternative is to use default methods, but that is Java 8 syntax, and not supported prior to Android SDK N, AFAIK.